### PR TITLE
Document deprecated keyfile and key args in nacl utils

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?
Add documentation about deprecate keyfile and key args in nacl utils. These args will continue to stay to ensure backwards compatibility but documentation will make its clear what the preferred method is.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/49427

### Previous Behavior
`keyfile` and `key` arguments included a deprecation notice
### New Behavior
`keyfile` and `key` will not see a deprecation notice. Users will need to use `sk_file` instead of `keyfile` and `sk` instead of `key` in accordance with the docs.

### Tests written?
Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: requirements/static/py3.5/linux.txt [hunk 1]
```diff
@@ -10,9 +10,7 @@ asn1crypto==0.24.0        # via cryptography
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5        # via tornado
 backports.functools-lru-cache==1.5  # via cheroot
-backports.ssl-match-hostname==3.7.0.1  # via websocket-client
```
Comment (Ch3LL) [inline requirements/static/py3.5/linux.txt]: @s0undt3ch is this normal for pre-commit to remove this? seems like we would want this backports. maybe i ran something wrong?

Comment (s0undt3ch) [reply requirements/static/py3.5/linux.txt]: It's the second time these changes happen.
Can you please try `pre-commit clean` and run it again?
I know you're on Manjaro, and you probably installed to the system python, that means 3.8.
Can you try the above and confirm the other questions please?

Comment (Ch3LL) [reply requirements/static/py3.5/linux.txt]: i think i was actually using my python2 virtualenv, im running into errors on 3.8 though, what version of python are you running it successfully on?

Comment (s0undt3ch) [reply requirements/static/py3.5/linux.txt]: 3.5 or 3.6, can't remember.

File: requirements/static/linux.in [hunk 2]
```diff
@@ -16,6 +16,7 @@ jxmlease
 kazoo
 keyring==5.7.1
 kubernetes<4.0
+libnacl==1.4.0
```
Comment (dwoz) [inline requirements/static/linux.in]: We need to recompile the static requirements if we change these `.in` files

**Review by dwoz — CHANGES_REQUESTED** [review]

File: requirements/static/linux.in (hunk 2 above)
Comment (s0undt3ch) [reply requirements/static/linux.in]: Yup, it's waiting on me.

Comment (s0undt3ch) [reply requirements/static/linux.in]: Updated

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**